app/controllers/criteria_controller.rb

```diff
@@ -0,0 +1,32 @@
+class CriteriaController < ApplicationController
+
+  def new
+    @assignment = Assignment.find(params[:assignment_id])
+    @criterion_type = params[:criterion_type]
+    @criterion = get_class(@criterion_type).new
```

Comment:
I'm saying to pass in the full class name as the param, not just "flexible" or "rubric".